docs/design-proposals/node/Partial-Runtime-Readiness-Handling.md

```diff
@@ -0,0 +1,121 @@
+
+# Partial Runtime Readiness Handling in Arktos Node Agent
+
+## Motivation
+In Arktos, multiple runtime services can be registered on the node to support different workloads, i.e. containers, VMs 
+and bare-metal machines (for details, please refer to the "Multiple Runtime Services Support" design spec). With multiple 
+runtime services registered on a node, a runtime service can be added or removed, and the state of the service can be 
+ready or notReady. As a result, the node status and node readiness management need to be updated to support those
+situations on the node, and the pod placement logic at the scheduler needs to be changed to ensure POD was scheduled to 
+a node with desired runtime service.
+
+This doc describes the proposal of the partial runtime readiness handling feature, including:
+1. Node readiness criteria with multiple runtime services on nodes
+2. Node runtime readiness state as part of the nodeStatus
+3. New scheduler predicate function for runtimeReadiness check
+
+## Requirements, principles and terminologies
+### Primary runtime of a cluster
+A primary runtime is the runtime service used for Kubernetes system daemon pods or pod with kube-system namespace. it is 
+usually a container runtime service given that the daemon runs as a process on the node. 
+
+### Runtime service state and state changes
+1. A Runtime service can be added or removed from a node
+2. A runtime service state can be at Ready, NotReady or Unknown; where Ready means both network and compute ready for 
+the runtime; NotReady means one of them or neither of them are ready in a particular runtime service. Unknown is the intial
```

Comment:
What do this mean: "network and compute ready for the runtime"?

Reply:
A container runtime service readiness include both compute part and network part readiness.  this is to refer to the network readiness in the runtime service.